### PR TITLE
Implement comparison of getLogs api

### PR DESCRIPTION
**Pull request by JonoPrest**
Hi @ozgrakkurt,

Ok so I've made a that just filters on logs along with a compareable rpc query.

I think we can do something similar on different aspects of the api.

### CODE REVIEW

File: integration-test/src/compare-get-logs.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    gasUsed,
+    timestamp,
+    extraData,
+    mixHash,
+    nonce: ethers.toBeHex(nonce),
+    totalDifficulty: ethers.toBeHex(totalDifficulty),
+    size,
+    hash,
+  };
+};
```
Comment (JonoPrest) [inline integration-test/src/compare-get-logs.js:58]: Note when we create these comparable objects we need to normalise values which I'm doing as hex.

File: integration-test/src/eth-archive-queries.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    //         data: true,
+    //         index: true,
+    //         removed: true,
+    //         topics: true,
+    //         transactionHash: true,
+    //         transactionIndex: true,
+    //       },
+    //     },
+    //   },
+    // ],
```
Comment (JonoPrest) [inline integration-test/src/eth-archive-queries.js:132]: I've removed this section of the query params and focusing on building a query just for logs along with the transactions/blocks that come back with those.

Will need to build another query to use rpc to get data in the same way that the transaction filters do on the eth archive api

**Review by ozgrakkurt — DISMISSED** [review]
LGTM, thank you!. Added some comments about the things I'm not sure of.

File: integration-test/src/eth-archive-queries.js (hunk 2 above)
Comment (ozgrakkurt) [reply integration-test/src/eth-archive-queries.js:132]: makes sense

**Comment by JonoPrest** [comment]
Hey @ozgrakkurt, I will fix the typo and add a few more comments.

To answer your question -> toBeHex works with any format I've tried. Number, strings, BigInt, hex strings. The main thing is that JSON rpc needed unpadded hex values (0x1 not 0x01) and returns either an integer or hex in unpadded format. Where as skar/eth-archive uses normalized padded format (0x01). So the only way to compare them is to normalize.

**Comment by JonoPrest** [comment]
@ozgrakkurt I've added some more comments and cleaned up a few things.

No logic changes.

**Review by ozgrakkurt — APPROVED** [review]
thanks! looks good to me

**Pull request merged**